src/script/lua_api/l_server.cpp

```diff
@@ -642,6 +642,25 @@ int ModApiServer::l_register_async_dofile(lua_State *L)
 	return 1;
 }
 
+// register_mapgen_script(path)
+int ModApiServer::l_register_mapgen_script(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	std::string path = readParam<std::string>(L, 1);
+	CHECK_SECURE_PATH(L, path.c_str(), false);
+
+	// Find currently running mod name (only at init time)
+	lua_rawgeti(L, LUA_REGISTRYINDEX, CUSTOM_RIDX_CURRENT_MOD_NAME);
```

Comment:
This mod name is used for check for insecure environment, if I understand correctly.

But this means that if a trusted mod A calls a function from another mod B, B can in this function register their own script under the name of A, and can hence request insec env.